format.js

```diff
@@ -0,0 +1,58 @@
+// consider this a warning about getting obsessive about optimization
+
+var utilformat = require('util').format
+
+
+function format (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
+  if (a16 === undefined) {
+    if (a15 === undefined) {
+      if (a14 === undefined) {
+        if (a13 === undefined) {
+          if (a12 === undefined) {
+            if (a11 === undefined) {
+              if (a10 === undefined) {
+                if (a9 === undefined) {
+                  if (a8 === undefined) {
+                    if (a7 === undefined) {
+                      if (a6 === undefined) {
+                        if (a5 === undefined) {
+                          if (a4 === undefined) {
+                            if (a3 === undefined) {
+                              if (a2 === undefined) {
+                                if (a1 === undefined)
+                                  return undefined
+                                return utilformat(a1)
+                              }
+                              return utilformat(a1, a2)
+                            }
+                            return utilformat(a1, a2, a3)
+                          }
+                          return utilformat(a1, a2, a3, a4)
+                        }
+                        return utilformat(a1, a2, a3, a4, a5)
+                      }
+                      return utilformat(a1, a2, a3, a4, a5, a6)
+                    }
+                    return utilformat(a1, a2, a3, a4, a5, a6, a7)
+                  }
+                  return utilformat(a1, a2, a3, a4, a5, a6, a7, a8)
+                }
+                return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9)
+              }
+              return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10)
+            }
+            return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11)
+          }
+          return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12)
+        }
+        return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13)
+      }
+      return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14)
+    }
+    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15)
+  }
+  return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16)
+}
+
+
+module.exports = format
```

Comment:
I feel like
```js
function format (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
  if (a16 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16)
  if (a15 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15)
  if (a14 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14)
  if (a13 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13)
  if (a12 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12)
  if (a11 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11)
  if (a10 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10)
  if (a9 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8, a9)
  if (a8 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7, a8)
  if (a7 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6, a7)
  if (a6 !== undefined)
    return utilformat(a1, a2, a3, a4, a5, a6)
  if (a5 !== undefined)
    return utilformat(a1, a2, a3, a4, a5)
  if (a4 !== undefined)
    return utilformat(a1, a2, a3, a4)
  if (a3 !== undefined)
    return utilformat(a1, a2, a3)
  if (a2 !== undefined)
    return utilformat(a1, a2)
  if (a1 !== undefined)
    return utilformat(a1)
  return undefined
}
```
Is a much cleaner way to write this. Not sure there would be a perf difference, though.